### ISSUE TITLE
Support omitting 'return' keyword

### ISSUE THREAD

**Issue by sgade** [vinivendra/Gryphon#72]
**Is your feature request related to a problem? Please describe.**

[Swift 5.1 introduced the ability](https://www.swiftbysundell.com/tips/omitting-the-return-keyword/) to omit the `return` keyword, if functions only contain a single expression.
This is currently not translated and leads to a missing `return` keyword in Kotlin.

**Describe the solution you'd like**

Ideally, if a function has only one expression that returns a value (so not `Void`), it should be preceded with a `return` if there is none.

**Describe alternatives you've considered**

Forcing a Swift user to add `return` statements where they are not necessary would be the alternative, and current, solution. This works but decreases the convenience in using Gryphon.

**Additional context**

Something like this:

```Swift
func example() -> String {
    "Hello, World"
}
```

should be translated to:

```Kotlin
internal fun example(): String {
   return "Hello, World"
}
```

(in Gryphon 0.9.0, the `return` keyword is missing)

**Comment by vinivendra** [comment]
Oh, I totally thought omitting the `return` would work there. Maybe the idiomatic translation of a single-expression function would be something more like this?

```` kotlin
internal fun example(): String = "Hello, World"
````

This should be a pretty easy fix in the `KotlinTranslator`: just check if the function's `statements` array contains a single expression, then print the ` = ` version if they are.

**Comment by sgade** [comment]
You're right, the example you gave is even better in Kotlin. 👍 

So this change would involve _not_ printing an opening bracket (`KotlinTranslator.swift` at 673 and 697) and then outputting the single expression in line 731, right? (Also should not add the closing brackets, of course)

**Comment by vinivendra** [comment]
Yeah, I think that could work. You would also need to add the `=` instead of the `{`, and make sure not to add the `}` at line 738.

**Comment by vinivendra** [comment]
@sgade I usually release a new version of Gryphon on Wednesdays, so I’ll probably start working on some of these issues before tomorrow evening. Let me know if there are any issues you’d like to fix yourself (like maybe this one) and I can skip them to go fix other stuff 😉

**Comment by sgade** [comment]
Thanks for the heads-up! With my available resources at the moment I'd like to start working on this particular issue.
I'll send you the PR once I have something working.

**Comment by sgade** [comment]
I have created a PR that involves functions.

Swift also supports omitting the `return` keyword in computed properties. I would tackle that once the PR is merged and I have a better understanding on how to work on the codebase. :)

**Comment by sgade** [comment]
Created that second PR. With that, I guess this issue would be done.

Are there any other places that this should be supported?
Seems like there'd need to be a more generic way to implement this, though... 🤔

**Comment by vinivendra** [comment]
I'm pretty sure these are all the places this should be supported (since it was already done for closures). That said, there might be a better way to do this once SwiftSyntax support is done, since I'm pretty sure SwiftSyntax comes with a "single-expression" flag we can use.

I closed PR #80 early for the release, but this issue can be closed once the missing test is added.
